src/layout/DwindleLayout.cpp

```diff
@@ -811,8 +837,40 @@ void CHyprDwindleLayout::alterSplitRatio(CWindow* pWindow, float ratio, bool exa
 }
 
 std::any CHyprDwindleLayout::layoutMessage(SLayoutMessageHeader header, std::string message) {
-    if (message == "togglesplit")
+    std::replace(message.begin(), message.end(), ' ', ',');
+    const auto ARGS = CVarList(message);
+    if (ARGS[0] == "togglesplit") {
         toggleSplit(header.pWindow);
+    } else if (ARGS[0] == "preselect") {
+        std::string direction = ARGS[1];
+
+        switch (direction.front()) {
+            case 'u':
+            case 't': {
+                overrideDirection = OneTimeFocus::UP;
+                break;
+            }
+            case 'd':
+            case 'b': {
+                overrideDirection = OneTimeFocus::DOWN;
+                break;
+            }
+            case 'r': {
+                overrideDirection = OneTimeFocus::RIGHT;
+                break;
+            }
+            case 'l': {
+                overrideDirection = OneTimeFocus::LEFT;
+                break;
+            }
+            default: {
+                // any other character resets the focus direction
+                // needed for the persistent mode
+                overrideDirection = OneTimeFocus::NOFOCUS;
```

Comment:
log an error too

Reply:
With the config any other direction is used to reset the persistent override, should i perhaps make a specific character for that and log an error for everything else ? otherwise logging an error here would mean logging an error for an intended usecase.

Reply:
check for `.empty()` as I am not sure whether reading `.front()` on `""` is valid

Reply:
good catch, it's UB.
The rest is fine though?

Reply:
dk, fix it and I'll review

Reply:
the check for empty is already there now.